### ISSUE TITLE
ModuleNotFoundError: No module named 'graph_transformer'

### ISSUE THREAD

**Issue by xli458** [Bhanu068/LAGNN#1]
Hi, 

I tried to replicate the results in your paper, but I have the following issues:

(1) this package does not have the correct version, and thus failed when I ran `bash setup.sh`
Since I cannot install dgl==0.8.1, I used dgl==0.9.0 instead, which is the closest version I could find. 

(2) With the above, I cannot import the following:

`from graph_transformer import GraphTransformerNet`

Could you kindly point out why? Thanks.

Regards,
David

**Comment by Bhanu068** [comment]
Hello David,

1) Could you provide more details regarding the error you faced?

2) You can remove the line "from graph_transformer import GraphTransformerNet" and run the code. GraphTransformerNet is not necessary. I will soon update the file.

Best regards,
Bhanu Prakash

**Comment by xli458** [comment]
Hi Bhanu,

I got your code running now. Thanks.

However, after reading your paper again, I have some comments.

(1) In your table 1, you reported the results of relation extraction (RE)
for various transformer models, and I think the results are problematic.
[1] reported the RE results for LayoutXLM (BASE), and it is 0.5483. We have
implemented the LayoutLMv3 (base) RE head, and got results around F1 0.56+,
using unilm repo (https://github.com/microsoft/unilm).
Your paper reported LayoutLMv3 RE can reach F1 0.805, which is
significantly higher.
I have come across a similar issue concerning the BROS [2], where they
reported a RE score as high as 71.46 for their base model. We found that
they assumed that each token has at most one in-coming relation in their
implementation. This assumption is not valid for the Funsd dataset. Below
is a snippet from image “0001476912.png” in the Funsd dataset:
[image: image.png]
*Please double check whether your implementation has the same issue. *

(2) You do not use any textual information for entity extraction, but only
layout. This makes me feel how general your solution will be for general
information extraction like on a CORD dataset.
https://github.com/clovaai/cord
Maybe a better solution will design a GNN head on top of a transformer
model like LayoutLMv3.

Regards,
David



[1] Xu, Y., Xu, Y., Lv, T., Cui, L., Wei, F., Wang, G., . . . Zhou, L.
(2021). LayoutXLM: Multimodal Pre-training for Multilingual Visually-rich
Document Understanding. *Proceedings of the 59th Annual Meeting of the
Association for Computational Linguistics and the 11th International Joint
Conference on Natural Language Processing* (pp. 2579–2591). Association for
Computational Linguistics.

[2] Hong, T., Kim, D., Ji, M., Hwang, W., Nam, D., & Park, S. (2021). BROS:
A Pre-trained Language Model for Understanding Texts in Document.
*International
Conference on Learning Representations.*

**Comment by Bhanu068** [comment]
@xli458 

In Table 1, we conducted a comparative analysis between our model LAGNN and other state-of-the-art (SOTA) models, employing our unique approach to Relation Extraction for a fair assessment. Our approach involved utilizing specific relation labels, namely "question-answer," "proximate_V," "proximate_H," "no-link," "header-question," and "same-entity." Conversely, existing methods employed different techniques for relation extraction. To gain a better understanding of their methodologies, I recommend referring to the respective papers. These differences in approach account for the differences in performance.

Furthermore, our primary objective was to develop a technique that extracts word or entity relations solely based on layout information, without relying on textual data. By doing so, we aimed to achieve generalizability across diverse forms and documents, regardless of the language used. Our method leverages relative distances between words as edge features, enabling its applicability to different forms and documents. Notably, the spacing between word pairs and the distance between sections or headers tend to remain relatively consistent across various forms and documents, particularly those generated by computers.
